### PR TITLE
chore(runtime): internalize pipe fd for ext/node

### PR DESCRIPTION
**Pull request by littledivy**

### CODE REVIEW

File: runtime/js/40_process.js [hunk 1]
```diff
@@ -30,6 +30,7 @@ import {
   ReadableStreamPrototype,
   writableStreamForRid,
 } from "ext:deno_web/06_streams.js";
+import { setPipeFdCallback } from "ext:deno_node/internal/child_process.ts";
```
Comment (littledivy) [inline runtime/js/40_process.js:33]: Not sure why snapshot fails: https://github.com/denoland/deno/actions/runs/7204945647/job/19627152677?pr=21570#step:30:137

Any ideas?

Comment (bartlomieju) [reply runtime/js/40_process.js:33]: It's probably a circular dependency between `ext:deno_node` and `ext:runtime` that V8 can't handle. Try using `internals` namespace to store this function.

File: ext/node/polyfills/internal/child_process.ts [hunk 2]
```diff
@@ -45,6 +45,7 @@ import { getValidatedPath } from "ext:deno_node/internal/fs/utils.mjs";
 import process from "node:process";
 
 const core = globalThis.__bootstrap.core;
+const internals = globalThis.__bootstrap.internals;
```
Comment (bartlomieju) [inline ext/node/polyfills/internal/child_process.ts:48]: Can you rewrite this to an import?

**Review by bartlomieju — APPROVED** [review]

File: ext/node/polyfills/internal/child_process.ts [hunk 3]
```diff
@@ -43,6 +44,7 @@ import {
 import { kEmptyObject } from "ext:deno_node/internal/util.mjs";
 import { getValidatedPath } from "ext:deno_node/internal/fs/utils.mjs";
 import process from "node:process";
+// HEAD
```
Comment (bartlomieju) [inline ext/node/polyfills/internal/child_process.ts:47]: Leftover from merge conflicts?

**Pull request merged**